assert/assertions.go

```diff
@@ -2161,24 +2161,46 @@ func NotErrorAs(t TestingT, err error, target interface{}, msgAndArgs ...interfa
 		return true
 	}
 
-	chain := buildErrorChainString(err)
+	chain := buildErrorChainString(err, true)
 
 	return Fail(t, fmt.Sprintf("Target error should not be in err chain:\n"+
-		"found: %q\n"+
+		"found: %T\n"+
```

Comment:
I should have noticed this the first time I reviewed but we need to dereference `target` once before formatting it here otherwise the expected type has an extra `*` in front of it.

We can do that with `reflect.ValueOf(target).Elem().Type()` which is safe enough because the `errors.As` call above would already panic if target is not a pointer.